resources/Microsoft.VSCode.DSC/Microsoft.VSCode.DSC.psm1

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum VSCodeEnsure
+{
+    Absent
+    Present
+}
+
+#region DSCResources
+[DSCResource()]
+class VSCodeExtension
```

Comment:
Recommend adding a static property and method for determining whether VS Code is installed/available for testing and simplified maintenance - the property should just hold the value for the check, and have it set in the static constructor.

Reply:
I created a new function called `Get-VSCodeCLIPath` that does two things,:
1. check the existence of VSCode installation
2. returns the path of the code.cmd file. 

I call this at the beginning of the module file and assign it to a local variable to by used by `Invoke-VSCode`. The reason I did it this way is so that we only have to determine the code.exe path once and all the functions that interact with the cli could be utilized by future DSC resources. I felt that a static variable was not entirely necessary because almost everything is driven though the interaction with the code.exe cli so checking for its existence should be a requirement.